### ISSUE TITLE
Multiple inconsistencies in array indexes

### ISSUE THREAD

**Issue by rebcabin** [lcompilers/lpython#2113]
Some of these problems may already be reported. Let this Issue serve as a MVE for multiple problems, explained in comments in the code:

```python
from lpython import (i8, # i32, i64, f32, f64,
                     dataclass
                     )
from numpy import (empty, sqrt, float32, float64,
                   int8, int32, array, # ndarray
                   )


@dataclass
class Foo:
    a : i8[4]

# uncomment to get error

# /Users/brian/CLionProjects/lpython/src/bin/python /Users/brian/CLionProjects/lpython/lasr/LP-pycharm/issue2108.py
# syntax error: Token 'foo' (of type 'identifier') is unexpected here
#   --> /Users/brian/CLionProjects/lpython/lasr/LP-pycharm/issue2108.py:17:1
#    |
# 17 | foo : Foo = Foo()
#    | ^^^

# a : i8[4]
# print(a[-1]

# No Error! BUT, negative index should produce error!

foo : Foo = Foo()
foo.a = empty(4, dtype=int8)

print(foo.a[-1])

# no error:
b : list[i8] = empty(4, dtype=int8)
# uncomment to get partly good, partly bad negative-index error.
# The reported range is bad (could be a duplicate)

# /Users/brian/CLionProjects/lpython/src/bin/python /Users/brian/CLionProjects/lpython/lasr/LP-pycharm/issue2108.py
# 0
# IndexError: List index is out of range. Index range is (0, -1), but the given index is -1

# print(b[-1])
```
